play.rb

```diff
@@ -0,0 +1,100 @@
+$LOAD_PATH << '.'
+require 'deck'
+require 'dealer'
+require 'player'
+
+def check_bust(player, dealer)
+  if player.total > 21
+    puts 'PLAYER BUST!! Dealer wins! '\
+    "Player total: #{player.total}, Dealer total: #{dealer.total}"
+    end_game
+  end
+  if dealer.total > 21
+    puts 'DEALER BUST!! Player wins! '\
+    "Player total: #{player.total}, Dealer total: #{dealer.total}"
+    end_game
+  end
+end
+
+def check_win(player, dealer)
+  d = 21 - dealer.total # How close dealer total is to 21
+  pl = 21 - player.total # How close player total is to 21
+  puts 'TIE GAME!' if d == pl
+  puts 'DEALER WINS! '\
+  "Player total: #{player.total}, Dealer total: #{dealer.total}" if d < pl
+  puts 'PLAYER WINS! '\
+  "Player total: #{player.total}, Dealer total: #{dealer.total}" if d > pl
+end
+
+def end_game
+  puts 'Thanks for playing! Seeya next time'
+  exit!
+end
+
+# Game starts here...
+puts 'Welcome to Blackjack!'
+deck = Deck.new
+
+p1 = Player.new
+p1.hand = [deck.game_deck[0], deck.game_deck[1]]
+p1.total = p1.calculate_total
+
+dealer = Dealer.new
+dealer.hand = [deck.game_deck[2], deck.game_deck[3]]
+dealer.total = dealer.calculate_total
+
+puts 'Cards in your hand: '
+puts p1.hand.each(&:to_s)
+puts "Your total is: #{p1.total}"
+
+puts "The dealer has: \nX"
+puts dealer.hand.drop(1).each(&:to_s)
+
+user_response = ''
+
+# Keeping track of cards in the deck, 4 have already been dealt
+card_num = 4
+
+# Continue asking until player stands
+until user_response.casecmp('stand') == 0
+  puts 'Hit or stand?'
+  user_response = gets.chomp
+
+  if user_response.casecmp('hit') == 0
+    puts 'You hit!'
+    p1.hand << deck.game_deck[card_num]
```

Comment:
I'd add here that you often hear talk of the 'law of demeter' in contexts such as this, which put simply states that any given class shouldn't know about anything more than the surface level of any other class. By manipulating the hand of the player here, you've gone 2 levels in: `player.hand.<<(some_card)` (yep, `<<` is a method in ruby!). A LoD-friendly refactor here could implement this interaction like `p1.hit!(deck.draw!)`, where the implementation of `hit!` is up to the player, and the implementation of `draw!` is up to the deck. I've made both of these bang methods because they manipulate internal state, but I can't speak for everyone in that preference.

I'd like to point out another difference in that code as well, `deck.draw!` vs `deck.game_deck[card_num]` -- the former can be implemented to shift the card off of the deck array, which models a real life game better than relying on a counter to know which card to give the player. Once a card has been drawn, it's no longer available to anyone else in the game, and a busted counter can't change that. It's a subtle distinction, but it's one less moving part.